browser/resources/settings/brave_leo_assistant_page/memory_section.html

```diff
@@ -0,0 +1,316 @@
+    align-items: center;
+    gap: var(--leo-spacing-s);
+    color: var(--leo-color-systemfeedback-error-text);
+    font: var(--leo-font-small-regular);
+  }
+  .error-message leo-icon {
+    color: var(--leo-color-systemfeedback-error-icon);
+  }
+</style>
+
+<div class="settings-box header">
+  <div class="flex">
+    <div class="main-header">$i18n{braveLeoAssistantYourMemoriesTitle}</div>
+    <div class="label secondary">
+      <span>$i18n{braveLeoAssistantYourMemoriesDesc} $i18nRaw{braveLeoAssistantMemoryLearnMore}</span>
+    </div>
+  </div>
+</div>
+
+<div class="list-container">
+  <settings-toggle-button id="userMemoryEnabled"
+      icon="database"
+      pref="{{prefs.brave.ai_chat.user_memory_enabled}}"
+      label="$i18n{braveLeoAssistantUserMemoryEnabledLabel}">
+  </settings-toggle-button>
+
+  <template
+    is="dom-if"
+    if="[[!hasMemories_(memoriesList_)]]"
+  >
+    <div class="empty-memories-table">
+      <div class="empty-memories-empty-state">
+        <div class="empty-memories-title">$i18n{braveLeoAssistantMemoryListEmptyTitle}</div>
+        <div class="empty-memories-desc">
+          $i18n{braveLeoAssistantMemoryListEmptyDescription}
+        </div>
+      </div>
+    </div>
+  </template>
+
+  <template
+    is="dom-if"
+    if="[[hasMemories_(memoriesList_)]]"
+  >
+    <div class="list">
+      <template
+        is="dom-repeat"
+        items="[[memoriesList_]]"
+      >
+        <div class="memory">
+          <div class="memory-info">
+            <div class="label">[[item]]</div>
+          </div>
+          <div class="memory-actions">
+            <cr-icon-button
+              on-click="handleEdit_"
+              class="size-20"
+              iron-icon="edit-pencil"
+            ></cr-icon-button>
+            <cr-icon-button
+              on-click="handleDelete_"
+              class="size-20"
+              iron-icon="trash"
+            ></cr-icon-button>
+          </div>
+        </div>
+      </template>
+    </div>
+  </template>
+</div>
+
+<div class="action-box">
+  <leo-button
+    on-click="handleAddNewMemory_"
+    kind="outline"
+    class="action-button">
+    $i18n{braveLeoAssistantAddNewMemoryButtonLabel}
+  </leo-button>
+  <leo-button
+    on-click="handleDeleteAllMemories_"
+    kind="plain"
+    class="action-button delete-all-button"
+    isDisabled$="[[!hasMemories_(memoriesList_)]]">
+    $i18n{braveLeoAssistantDeleteAllMemoriesConfirmationTitle}
+  </leo-button>
+</div>
+
+<template
+  is="dom-if"
+  if="[[showMemoryDialog_]]"
+  restamp
+>
+  <cr-dialog id="memoryDialog" show-on-attach show-close-button on-close="onDialogCancel_">
+    <div slot="title">[[getDialogTitle_(editingMemoryItem_)]]</div>
+    <div slot="body">
+      <leo-input required id="memoryInput" type="text" value="[[editingMemory_]]" on-change="onDialogInput_" on-input="onDialogInput_" showErrors$="[[isMemoryInputError_]]" placeholder="[[getInputPlaceholder_(editingMemory_)]]" style="width:100%">
```

Comment:
is it intentional that you've bound `on-input` and `on-change`? `on-input` should fire every keystroke, while I think `on-change` will only fire when it loses focus.

Probably you just want `on-input`